app/src/main/java/com/example/calculator/MainActivity.kt

```diff
@@ -2,179 +2,84 @@ package com.example.calculator
-            {
-                if(canAddDecimal)
-                    workingTextView.append(view.text)
-
-                canAddDecimal = false
-            }
-            else
-                workingTextView.append(view.text)
-
-            canAddOperation = true
-        }
-    }
-
-    fun operationAction(view: View)
-    {
-        if(view is Button && canAddOperation)
-        {
-            workingTextView.append(view.text)
-            canAddOperation = false
-            canAddDecimal = true
-        }
-    }
-
-    fun allClear(view: View)
-    {
-        workingTextView.text = ""
-        resultsTextView.text = ""
-    }
-
-    fun delAction(view: View)
-    {
-        val length = workingTextView.length()
-        if(length > 0)
-            workingTextView.text = workingTextView.text.subSequence(0, length - 1)
-    }
-
-    fun equalsAction(view: View)
-    {
-        resultsTextView.text = calculateResults()
-        resultsTextView.text = calculateResults()
-    }
-
-    private fun calculateResults(): String
-    {
-        val digitsOperators = digitsOperators()
-
-        if(digitsOperators.isEmpty())
 
-            return ""
+        workingTextView = findViewById(R.id.workingTextView)
+        val workingTextView: TextView = findViewById(R.id.workingTextView)
 
-        val timesDivision = timesDivisionCalculate(digitsOperators)
-
-        if(timesDivision.isEmpty())
-
-            return ""
-
-        val result = addSubtractCalculate(timesDivision)
-
-        return result.toString()
-    }
-
-    private fun addSubtractCalculate(oldList: MutableList<Any>): Float
-    {
-        var result = oldList[0] as Float
-
-        for(i in oldList.indices)
-        {
-            if(oldList[i] is Char && i != oldList.lastIndex)
-            {
-                val operator = oldList[i]
-                val nextDigit = oldList[i + 1] as Float
-                if (operator == '+')
-                    result += nextDigit
-                if (operator == '-')
-                    result -= nextDigit
-            }
+        numberButtons = arrayOf(number9,number8,number7,number6, number5,number4,number3,number2,number1,number0)
+        for (i in numberButtons){
+            i.setOnClickListener {numberButtonClick(i)}
         }
-
-        return result
+        operatorButtons = listOf(DIV,MUL,SUB,ADD,CLEAR)
+        for (i in operatorButtons){i.setOnClickListener {operatorButtonClick(i)}}
+        equal.setOnClickListener {buttonEqualClick()}
+        CLEAR.setOnClickListener {buttonClearClick()}
     }
-
-    private fun timesDivisionCalculate(oldList: MutableList<Any>): MutableList<Any>
-    {
-        var list = oldList
-        while (list.contains('x') || list.contains('/'))
-        {
-            list = calcTimesDiv(list)
+    private fun buttonClearClick(){
+        val result = when(operator){
+            Operator.CLEAR -> 0
+            else -> 0
         }
-        return list
+        strNumber.clear()
```

Comment:
strNumber is cleared and appended here just to be used to fill workingTextView
Consider if it is needed at all, and if it is needed in this exact place
Maybe it's value could be directly assigned to workingTextView text